web/src/message_view.js

```diff
@@ -661,14 +661,23 @@ export function show(raw_terms, opts) {
                     validate_filter_topic_post_fetch:
                         filter.requires_adjustment_for_moved_with_target,
                     cont() {
-                        if (
-                            !filter.requires_adjustment_for_moved_with_target &&
-                            filter.has_operator("with")
-                        ) {
+                        if (filter.narrow_requires_hash_change) {
                             // We've already adjusted our filter via
                             // filter.try_adjusting_for_moved_with_target, and
                             // should update the URL hash accordingly.
                             update_hash_to_match_filter(filter, "retarget topic location");
+                            // Since filter is updated, we need to handle various things
+                            // like updating the message view header title, unread banner
+                            // based on the updated filter.
+                            handle_post_message_list_change(
+                                id_info,
+                                message_lists.current,
+                                opts,
+                                select_immediately,
+                                select_opts,
+                                then_select_offset,
+                            );
+                            filter.narrow_requires_hash_change = false;
```

Comment:
When filter is set up `narrow_requires_hash_change` is set to false by constructor. It is set to false later here (in case its true). We can't do it in `update_hash_to_match_filter` because it is required to rebuild the `sorted_term_types` to update the message view header title, through the `handle_post_view_change` in case the hash is updated.

Reply:
OK, this likely will want an explanatory comment to make the reasoning visible to future readers of this code.

Reply:
I guess maybe the existing comment is good enough.